cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -772,32 +779,36 @@ TEST_F(ScalarTemporalTest, TestZoned1) {
                         {"iso_year": 2008, "iso_week": 52, "iso_day_of_week": 6},
                         {"iso_year": 2008, "iso_week": 52, "iso_day_of_week": 7},
                         {"iso_year": 2011, "iso_week": 52, "iso_day_of_week": 6}, null])");
-  auto quarter = "[4, 1, 4, 2, 4, 4, 4, 4, 4, 1, 1, 4, 4, 4, 4, 4, null]";
-  auto hour = "[14, 13, 15, 18, 15, 16, 17, 18, 19, 21, 22, 23, 0, 14, 14, 15, null]";
-  auto minute = "[30, 53, 41, 3, 35, 40, 45, 50, 55, 0, 5, 10, 15, 30, 30, 32, null]";
+    auto quarter = "[4, 1, 4, 2, 4, 4, 4, 4, 4, 1, 1, 4, 4, 4, 4, 4, null]";
+    auto hour = "[14, 13, 15, 18, 15, 16, 17, 18, 19, 21, 22, 23, 0, 14, 14, 15, null]";
+    auto minute = "[30, 53, 41, 3, 35, 40, 45, 50, 55, 0, 5, 10, 15, 30, 30, 32, null]";
+    if (timezone == "-09:30") {
+      minute = "[30, 53, 29, 3, 35, 40, 45, 50, 55, 0, 5, 10, 15, 30, 30, 32, null]";
```

Comment:
Hmm, why is the third value different from above? Can you add a comment explaining this?

Reply:
Added:
```cpp
      // Prior to October 1st 1912 Pacific/Marquesas was on solar time (probably)
      // and is on +09:30 since.
```